### PR TITLE
Update for upstream changes

### PR DESCRIPTION
**Pull request by reem**
Other deps (rustc-serialize, regex) are still broken, so don't merge yet.

### CODE REVIEW

**Comment by seanmonstar** [comment]
Will need a rebase since I merged in @pyfisch's PR.

**Comment by reem** [comment]
`regex` and `rustc-serialize` are *still* broken despite open PRs, so this doesn't really fix anything yet.

**Comment by martinsp** [comment]
`regex` and `rustc-serialize` are updated now and compiles on rustc 0.13.0-nightly (c89417130 2015-01-02 21:56:13 +0000)

**Comment by seanmonstar** [comment]
I just bumped Travis, and serialize errores again...

**Comment by seanmonstar** [comment]
Oh, seems something is explicitly depending on an older version.

**Comment by seanmonstar** [comment]
@reem just checked again, Travis errors in hyper now.

**Comment by reem** [comment]
Trying to fix... So many errors...

**Comment by reem** [comment]
Trying to fix but getting bizarre errors when trying to implement Deref and DerefMut, looks like an associated types error.

**Comment by Gankro** [comment]
@reem Anything I can do to help with this fix?

**Comment by reem** [comment]
@Gankro do you have push rights/crates.io rights for the other crates in rust-lang (regex, log, etc.)?

**Comment by Gankro** [comment]
Sadly not. :(

File: src/header/mod.rs [hunk 1]
```diff
@@ -433,7 +439,7 @@ impl fmt::Show for Item {
             None => match self.raw {
                 Some(ref raw) => {
                     for part in raw.iter() {
-                        try!(fmt.write(part.as_slice()));
+                        try!(write!(fmt, "{}", part.as_slice()));
```
Comment (reem) [inline src/header/mod.rs]: @seanmonstar please confirm this is correct

Comment (reem) [inline src/header/mod.rs]: Just ran some of the examples to sanity check and it's fine, so I guess this works as intended.

**Comment by reem** [comment]
@seanmonstar r?

**Pull request merged**